DatadogInternal/Sources/Models/SessionReplay/PrivacyLevel+SessionReplay.swift

```diff
@@ -0,0 +1,81 @@
+ */
+
+#if os(iOS)
+import UIKit
+
+// MARK: - DatadogExtension for UIView
+
+/// Extension on `DatadogExtension` to provide access to `SessionReplayOverride` for any `UIView`.
+extension DatadogExtension where ExtendedType: UIView {
+    /// Provides access to Session Replay override settings for the view.
+    /// Usage: `myView.dd.sessionReplayOverride.textAndInputPrivacy = .maskNone`.
+    public var sessionReplayOverride: SessionReplayOverrideExtension<ExtendedType> {
+        get {
+            return SessionReplayOverrideExtension(self.type)
+        }
+        set {}
+    }
+}
+
+// MARK: - Associated Keys
+
+private var associatedTextAndInputPrivacyKey: UInt8 = 3
+private var associatedImagePrivacyKey: UInt8 = 4
+private var associatedTouchPrivacyKey: UInt8 = 5
+private var associatedHiddenPrivacyKey: UInt8 = 6
+
+// MARK: - SessionReplayOverrideExtension
+
+/// `UIView` extension  to manage the Session Replay privacy override settings.
+public struct SessionReplayOverrideExtension<ExtendedType> {
+    private let view: ExtendedType
+
+    public init(_ view: ExtendedType) {
+        self.view = view
+    }
+
+    /// Text and input privacy override (e.g., mask or unmask specific text fields, labels, etc.).
+    public var textAndInputPrivacy: TextAndInputPrivacyLevel? {
+        get {
+            return objc_getAssociatedObject(view as AnyObject, &associatedTextAndInputPrivacyKey) as? TextAndInputPrivacyLevel
+        }
+        set {
+            objc_setAssociatedObject(view as AnyObject, &associatedTextAndInputPrivacyKey, newValue, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
+        }
+    }
+
+    /// Image privacy override (e.g., mask or unmask specific images).
+    public var imagePrivacy: ImagePrivacyLevel? {
+        get {
+            return objc_getAssociatedObject(view as AnyObject, &associatedImagePrivacyKey) as? ImagePrivacyLevel
+        }
+        set {
+            objc_setAssociatedObject(view as AnyObject, &associatedImagePrivacyKey, newValue, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
+        }
+    }
+
+    /// Touch privacy override (e.g., hide or show touch interactions on specific views).
+    public var touchPrivacy: TouchPrivacyLevel? {
+        get {
+            return objc_getAssociatedObject(view as AnyObject, &associatedTouchPrivacyKey) as? TouchPrivacyLevel
+        }
+        set {
+            objc_setAssociatedObject(view as AnyObject, &associatedTouchPrivacyKey, newValue, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
+        }
+    }
+
+    /// Hidden privacy override (e.g., mark a view as hidden, rendering it as an opaque wireframe in replays).
+    public var hiddenPrivacy: HiddenPrivacyLevel? {
+        get {
+            return objc_getAssociatedObject(view as AnyObject, &associatedHiddenPrivacyKey) as? HiddenPrivacyLevel
+        }
+        set {
+            objc_setAssociatedObject(view as AnyObject, &associatedHiddenPrivacyKey, newValue, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
+        }
+    }
```

Comment:
Maybe simple `Bool` is good enough for this? `.hidden` and `.none` options cause a little bit of confusion imo 🤔

Reply:
I thought of it. I agree that `.none` is a bit confusing, especially since it's used to represent an optional in Swift, which means customers will have to explicitly specify `HiddenPrivacyLevel.none` for the compiler to resolve it correctly. One alternative is to rename it to something like `.disabled`. Another possibility is for the `enum` to have a single case, `.hide` (and the privacy can be clearer by setting it to `nil` since it's optional in the SR extension).

The main advantage of using an `enum` over a `Bool` is that we can extend it later, in case we want to add other use cases (a point @ganeshnj raised at the time of writing the [RFC](https://datadoghq.atlassian.net/wiki/spaces/RUMP/pages/3901129961/RFC+-+Session+Replay+Fine+Grained+Masking+Overrides)). However, for the current use case, a `Bool` might make more sense.

And ideally we want iOS and Android to align (cc @3Ajonathanmos).

Reply:
Following our meeting, we decided to use a `Boolean` instead.